### DISCUSSION TITLE
Automatic message in chat

### DISCUSSION THREAD

**Discussion by MrAuronTV** [TwitchIO/TwitchIO#465] [Questions]
Hello

I would like create automatic message every x minute in twitch chat , can we do that with twitchio ?

Thanks

**Comment by chillymosh** [comment]
Yes, you can use the routines extension.

**Reply by MrAuronTV** [reply]
how ? in the example, we can just print a message in the console

**Reply by chillymosh** [reply]
```py
    @routines.routine(minutes=10)
    async def panda(self) -> None:
        channel = self.bot.get_channel("chillymosh")
        await channel.send("I am a panda")

    @panda.before_routine
    async def before_panda(self) -> None:
        await self.bot.wait_for_ready()
```

**Reply by MrAuronTV** [reply]
Thanks! i try this 👍

**Reply by MrAuronTV** [reply]
i've try your solution with other name but i have an issue

![image](https://github.com/user-attachments/assets/9f8338b1-b838-464d-9c82-0bce8efecdd8)

**Reply by chillymosh** [reply]
Firstly, you need to update your Python, 3.7 is EOL, outdated and should definitely not be used anymore.
The main issue though, is it looks like you may not understand how OOP / objects work in Python as the error indicates an issue with `self`. Check where you are performing this.

**Reply by MrAuronTV** [reply]
I've found ! thank you very much for the help

I called the function in the wrong place, I just had to call it in the init

self.your_def.start()

**Comment by MrAuronTV** [comment]
Hello

Automatic messages work well but messages are sent even when I'm not streaming, is this normal? Or can I make sure that messages aren't sent until I'm streaming?

**Comment by chillymosh** [comment]
Yes, this is normal. Same as any normal person on the website can send messages when a channel is not streaming.

If you want to only send routine messages when channel is online then you'll need to use eventsub as triggers.

**Comment by MrAuronTV** [comment]
Ok thanks i check that
